Code/Tools/FBuild/FBuildCore/Graph/TextFileNode.cpp

```diff
@@ -0,0 +1,149 @@
+// TextFileNode.cpp
+//------------------------------------------------------------------------------
+
+// Includes
+//------------------------------------------------------------------------------
+#include "TextFileNode.h"
+
+#include "Tools/FBuild/FBuildCore/BFF/Functions/Function.h"
+#include "Tools/FBuild/FBuildCore/FBuild.h"
+#include "Tools/FBuild/FBuildCore/FLog.h"
+#include "Tools/FBuild/FBuildCore/Graph/NodeGraph.h"
+#include "Tools/FBuild/FBuildCore/Graph/DirectoryListNode.h"
+
+#include "Core/Env/ErrorFormat.h"
+#include "Core/FileIO/FileIO.h"
+#include "Core/FileIO/FileStream.h"
+#include "Core/Math/Conversions.h"
+#include "Core/Strings/AStackString.h"
+#include "Core/Process/Process.h"
+
+// Reflection
+//------------------------------------------------------------------------------
+REFLECT_NODE_BEGIN( TextFileNode, Node, MetaName( "TextFileOutput" ) + MetaFile() )
+    REFLECT_ARRAY( m_TextFileInputStrings, "TextFileInputStrings", MetaNone() )
```

Comment:
I considered making TextFileInputStrings optional: it would just be a file with no contents.  However, I figured there was basically no situation where a 3-byte file wasn't just as good as an empty file, and it might be super frustrating if someone had set `TextFileInputstrings` or something which was just silently ignored and their file was empty.

Reply:
What does make me sad is that empty strings aren't valid in an array of strings, but that's not something I can control from here